cluster-essentials/pkg/client/install.go

```diff
@@ -20,28 +24,65 @@ var appplyResourcesFromManifests = appplyResourcesFromManifest
 const defaultBundleDir = "/tmp"
 const defaultTimeout = 15 * time.Minute
 
+var clusterEssentialPackages = []string{"kapp-controller", "secretgen-controller"}
+
 func validatePreInstall(restConfig *rest.Config) error {
-	// TODO implementation
-	return nil
+	clientset, _ := kubernetes.NewForConfig(restConfig)
+	nsList, _ := clientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	for _, n := range nsList.Items {
+		for _, packageName := range clusterEssentialPackages {
+			deploymentsClient := clientset.AppsV1().Deployments(n.Name)
+			result, getErr := deploymentsClient.Get(context.TODO(), packageName, metav1.GetOptions{})
+			if getErr != nil {
+				//TODO check for CRD if deployment is not present
+				// if Deplyoment and CRD is not present, continue the loop to check in other namespace 
+				continue
+			}
+			// If deployment is present, check for annotation
+			_, ok := result.Annotations["tanzu.vmware.cluster.essentia/package-name"]
```

Comment:
`tanzu.vmware.cluster.essentia/package-name`

This is not the right format for annotations. This should be `runtime.tanzu.vmware.com/<something>`.